### PR TITLE
[Regression] Make a redirection optional for the menu item alias

### PR DESCRIPTION
**Pull request by csthomas**
Pull Request for Issue https://github.com/joomla/joomla-cms/pull/22432#issuecomment-435829447

### Summary of Changes
Restore the default behaviour of the menu item alias to Joomla before version 3.9

Redirection is now optional and will not work on POST request.

### Testing Instructions
See comments at #22432

### Expected result
See comments at #22432

### Actual result
See comments at #22432

### Documentation Changes Required
New language strings.

### CODE REVIEW

File: libraries/src/Application/CMSApplication.php [hunk 1]
```diff
@@ -1075,7 +1075,10 @@ protected function route()
 
 		$active = $this->getMenu()->getActive();
 
-		if ($active !== null && $active->type === 'alias')
+		if ($this->input->getMethod() !== 'POST'
```
Comment (mbabker) [inline libraries/src/Application/CMSApplication.php]: This should really be flipped to explicitly only support GET.  In the off chance someone actually tries to use Joomla to perform PUT or DELETE requests, which will have the same fundamental issues as POST.

Comment (csthomas) [reply libraries/src/Application/CMSApplication.php]: Also HEAD
```suggestion
		if (in_array($this->input->getMethod(), array('GET', 'HEAD'), true))
```

Comment (csthomas) [reply libraries/src/Application/CMSApplication.php]: Thanks

File: administrator/language/en-GB/en-GB.com_menus.ini [hunk 2]
```diff
@@ -94,6 +94,8 @@ COM_MENUS_ITEM_DETAILS="Details"
 COM_MENUS_ITEM_FIELD_ALIAS_DESC="The alias is used in the URL when SEF is on."
 COM_MENUS_ITEM_FIELD_ALIAS_MENU_DESC="Menu Item to link to."
 COM_MENUS_ITEM_FIELD_ALIAS_MENU_LABEL="Menu Item"
+COM_MENUS_ITEM_FIELD_ALIAS_REDIRECT_DESC="If set to Yes then visitors will be redirected to the linked menu item."
+COM_MENUS_ITEM_FIELD_ALIAS_REDIRECT_LABEL="Use redirection"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_menus.ini]:
```suggestion
COM_MENUS_ITEM_FIELD_ALIAS_REDIRECT_LABEL="Use Redirection"
```

**Comment by sanderpotjer** [comment]
I have tested this item :white_check_mark: successfully on f354ab3786383195af917e0db37037c1ead03384<br /><br />@csthomas great, thanks a lot! I confirm this is restoring the "old" behaviour, and makes the redirect optional. Template style is hiding nicely when redirect is set to Yes. 

Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23278">issues.joomla.org/tracker/joomla-cms/23278</a>.</sub>

**Comment by csthomas** [comment]
I have changed the order of conditions in the if statement so that I do not check the type of request in all requests.

**Comment by phproberto** [comment]
I have tested this item :white_check_mark: successfully on 83de6dd396d7be3d8d42dddbbe0e90dbfba64d8e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23278">issues.joomla.org/tracker/joomla-cms/23278</a>.</sub>

**Comment by sanderpotjer** [comment]
I have tested this item :white_check_mark: successfully on 83de6dd396d7be3d8d42dddbbe0e90dbfba64d8e<br /><br />Also after the latest commit, thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23278">issues.joomla.org/tracker/joomla-cms/23278</a>.</sub>

**Comment by infograf768** [comment]
RTC after 2 good tests.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23278">issues.joomla.org/tracker/joomla-cms/23278</a>.</sub>

**Comment by csthomas** [comment]
I have created one more PR (https://github.com/joomla/joomla-cms/pull/23294) to allow 3rd party extensions/custom code to create a SEF url for alias links (menu item alias)

**Pull request merged**